Sources/swift-package-manager/SwiftPM.swift

```diff
@@ -22,14 +22,14 @@ let execName = (try? AbsolutePath(validating: firstArg).basenameWithoutExt) ??
 
 @main
 struct SwiftPM {
-    static func main() {
+    static func main() async {
```

Comment:
I think we had trouble with this change in the past, are we out of the wood on that now?

cc @neonichu

Reply:
AFAIR those were CI issues, but I don't see them reproduced now and this passes tests and runs well locally. I don't plan to cherry-pick this to 5.9, so we'll have enough time to investigate anything that could potentially go wrong.

Reply:
Windows is another question, I think? Supposedly async doesn't really work well there IIRC. cc @compnerd

Reply:
maybe also check the toolchain CI, ie try to create a toolchain with this PR, I remember we had SwiftPM green but it caused downstream issues. in any case, I think we should not merge this in the immediate timeframe to avoid 🔥 in a sensitive time